sound/soc/soc-core.c

```diff
@@ -865,8 +865,12 @@ static int soc_dai_link_sanity_check(struct snd_soc_card *card,
 		 * Defer card registration if codec component is not added to
 		 * component list.
 		 */
-		if (!soc_find_component(codec))
+		if (!soc_find_component(codec)) {
+			dev_err(card->dev,
+				"ASoC: codec component %s not found for link %s\n",
+				codec->name, link->name);
```

Comment:
I might be not quite right, but EPROBE_DEFER means some conditions for probing aren't satisfied yet, e.g. some other drivers haven't probed yet, so we delay probing. When a driver returns this code, it will be re-probed *every time* another probing succeeds. So, if in some set up you run onto this "error," with your patch you'll print this error message every time some other unrelated driver probes successfully. Don't think this deserves a `dev_err()`, at most a `dev_dbg()`

Reply:
@lyakh Ack, I don't think we can use dev_err() here, but maybe dev_dbg() ? Currently debugging card registration failures is super painful ... if some conditions never show up, card probe just silently fails with no trace what was the missing component for which everybody is waiting.

Reply:
right, it should be dev_dbg() and we should add in the commit message that this comes direct user support where we can't figure out what's going on when the card registration fails despite successful probe of the platform driver.

Reply:
ack. done!